core/domain/user_services_test.py

```diff
@@ -653,6 +823,91 @@ def test_get_user_ids_by_role(self):
                 feconf.ROLE_ID_CURRICULUM_ADMIN)),
             set([user_ids[2], user_ids[3]]))
 
+    def test_update_user_bio(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        user_bio = 'new bio'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        pre_update_user_settings = user_services.get_user_settings(user_id)
+        self.assertNotEqual(pre_update_user_settings.user_bio, user_bio)
+
+        user_services.update_user_bio(user_id, user_bio)
+        user_settings = user_services.get_user_settings(user_id)
+
+        self.assertEqual(user_bio, user_settings.user_bio)
+
+    def test_update_preferred_language_codes(self):
+        language_codes = ['en']
+
+        user_id = user_services.create_new_user(
+            'someUser',
+            'user@example.com').user_id
+        user_services.update_preferred_language_codes(
+            user_id, language_codes)
+        user_settings = user_services.get_user_settings(user_id)
+
+        self.assertEqual(
```

Comment:
Missing pre-check. Please verify for all your tests that they contain a pre-check and a post-check (where relevant).

Reply:
I added pre-checks.